src/gcp/client.rs

```diff
@@ -640,9 +632,11 @@ impl GetClient for GoogleCloudStorageClient {
             request = request.query(&[("generation", version)]);
         }
 
-        if !credential.bearer.is_empty() {
-            request = request.bearer_auth(&credential.bearer);
-        }
+        request = request.with_bearer_auth(credential.as_deref());
+
+        // if !credential.bearer.is_empty() {
+        //     request = request.bearer_auth(&credential.bearer);
+        // }
```

Comment:
Reading this code, it looks like previously if `credential.bearer` was `""`, it would also skip signing the request, but only for `GET` requests. Should we maintain that functionality? I.e. should we change the diff below to

```rs
impl CredentialExt for HttpRequestBuilder {
    fn with_bearer_auth(self, credential: Option<&GcpCredential>) -> Self {
        match credential {
            Some(credential) => {
                if credential.bearer.is_empty() {
                    return self;
                }

                self.bearer_auth(&credential.bearer)
            }
            None => self,
        }
    }
}
```

Reply:
[`640f7f2` (#322)](https://github.com/apache/arrow-rs-object-store/pull/322/commits/640f7f21e82ce4cbc9402c0f9e7aa3884fff3dc5) implements this change, so we also now skip signing when bearer is `""`, to ensure this isn't a breaking change.